misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -274,81 +383,294 @@ public static Symbol findSymbol(List<TypeSymbol> fileContentSymbols) {
+                for (Map.Entry<String, Map<String, RecordFieldSymbol>> field : this.spreadFieldMap.entrySet()) {
+                    JsonObject spreadFieldDetails = new JsonObject();
+                    spreadFieldDetails.addProperty(ID, "dummy_id");
+                    spreadFieldDetails.addProperty(TYPE, "ballerina_type");
+                    spreadFieldDetails.add(PROPERTIES, rightRecordToJSON(field.getValue().values()));
+                    rightSchema.add(field.getKey(), spreadFieldDetails);
+                }
+            }
 
             rightRecordJSON.addProperty(SCHEMA, foundTypeRight);
             rightRecordJSON.addProperty(ID, "dummy_id");
             rightRecordJSON.addProperty(TYPE, "object");
             rightRecordJSON.add(PROPERTIES, rightSchema);
-        }
 
-        // Schema 2
-        TypeSymbol lftSymbol = ((TypeReferenceTypeSymbol) lftTypeSymbol).typeDescriptor();
-        if ("RECORD".equals(lftSymbol.typeKind().name())) {
-            RecordTypeSymbol leftSymbol = (RecordTypeSymbol) lftSymbol;
-            Map<String, RecordFieldSymbol> leftSchemaFields = leftSymbol.fieldDescriptors();
+            Map<String, Object> rightSchemaMap = new Gson().fromJson(rightSchema,
+                    new TypeToken<HashMap<String, Object>>() {
+                    }.getType());
+            this.isOptionalMap.clear();
+            generateOptionalMap(rightSchemaMap, foundTypeRight);
+
 
-            JsonObject leftSchema = (JsonObject) recordToJSON(leftSchemaFields.values());
+            // Schema 2
+            Map<String, RecordFieldSymbol> leftSchemaFields = symbolList.get(1).fieldDescriptors();
+            JsonObject leftSchema = (JsonObject) leftRecordToJSON(leftSchemaFields.values());
             leftRecordJSON.addProperty(SCHEMA, foundTypeLeft);
             leftRecordJSON.addProperty(ID, "dummy_id");
             leftRecordJSON.addProperty(TYPE, "object");
             leftRecordJSON.add(PROPERTIES, leftSchema);
-        }
-
 
+            Map<String, Object> leftSchemaMap = new Gson().fromJson(leftSchema,
+                    new TypeToken<HashMap<String, Object>>() {
+                    }.getType());
+            this.leftFieldMap.clear();
+            getLeftFields(leftSchemaMap, "");
+        }
         JsonArray schemas = new JsonArray();
         schemas.add(leftRecordJSON);
         schemas.add(rightRecordJSON);
-        return getMapping(schemas, context, foundTypeLeft, foundTypeRight);
+        return getMapping(schemas, context);
+    }
+
+    private JsonObject insertRestFields(JsonObject rightSchema, String foundTypeRight) {
+        // Added to get the rest fields
+        HashMap<String, String> tempRestFieldMap = new HashMap<>(this.restFieldMap);
+        Set<Map.Entry<String, String>> entrySet = tempRestFieldMap.entrySet();
+        for (Map.Entry<String, String> restField : entrySet) {
+            JsonObject fieldDetails = new JsonObject();
+            fieldDetails.addProperty(ID, "dummy_id");
```

Comment:
Avoid using the `dummy_` prefix and use something else

Reply:
removed `dummy_`